File: 287-find-the-duplicate-number/287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n = nums.size();
        
        for( int i = 0; i < n; i++ ) {
            while (nums[i] != i) {
                int d = nums[i];
                if( nums[i] != nums[d] )
                    swap(nums[i], nums[d]);
                else
                    break;
            }
        }
    
        for( int i = 0; i < n; i++)  {
            if (nums[i] != i)
                return nums[i];
        }
        
        return n;
    
    }
};

File: 287-find-the-duplicate-number/287-find-the-duplicate-number.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        for i in range(len(nums)):
            while nums[i] != i:
                d = nums[i]
                if nums[i] != nums[d]:
                    nums[i], nums[d] = nums[d], nums[i]
                else:
                    break
            for i in range(len(nums)):
                if nums[i] != i: 
                    return nums[i]
                

                
            
                
            
        
        